src/Common/src/System/Net/Logging/Logging.cs

```diff
@@ -0,0 +1,724 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Threading;
+
+namespace System.Net
+{
+    internal class Logging
+    {
+        private static volatile bool s_LoggingEnabled = true;
```

Comment:
@stephentoub /cc: @josguil, @vancem  Yes: by design all ETW tracing will be "enabled". 
I agree that we need to check that the log is enabled prior to performing all calculations. This was left out given that GlobalLog was previously only used when DEBUG was enabled. We'll follow the same pattern as with the Logging object.
Tracking this item with #2500.